src/test/java/com/faforever/api/map/MapControllerTest.java

```diff
@@ -0,0 +1,92 @@
+package com.faforever.api.map;
+
+import com.faforever.api.authentication.AuthenticationService;
+import com.faforever.api.config.FafApiProperties;
+import com.faforever.api.config.TestWebSecurityConfig;
+import com.faforever.api.player.PlayerRepository;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.io.ByteStreams;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.Import;
+import org.springframework.mock.web.MockMultipartFile;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+
+import java.io.InputStream;
+
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.fileUpload;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(MapsController.class)
+@Import(TestWebSecurityConfig.class)
+public class MapControllerTest {
+
+  @Autowired
+  private MockMvc mvc;
+  @MockBean
+  private PlayerRepository playerRepository;
+  @MockBean
+  private MapRepository mapRepository;
+  @MockBean
+  private MapService mapService;
+  @MockBean
+  private FafApiProperties fafApiProperties;
+  @MockBean
+  private AuthenticationService authenticationService;
+  @MockBean
+  private ObjectMapper objectMapper;
```

Comment:
Most of the fields above are, as IntelliJ tells you, unused. Remove them.
Where it complains about "never assigned", use:
![image](https://cloud.githubusercontent.com/assets/1864183/22424821/9d0cfc14-e6f8-11e6-86a9-0ed4454b4fb7.png)